It is a type of higher order components that wraps other components.
When the other components or children of an ErrorBoundary component throw an error,
we catch it in the ErrorBoundary component by implementing 'componentDidCatch' method
in the ErrorBoundary component.

In the render of the ErrorBoundary component,
```js
	constructor(props) {
		super(props);
		this.state = { hasError: false };
	}

	static getDerivedStateFromError(error) {
		return {
			hasError: true,
		};
	}

	componentDidCatch = (err, info) => {
		// pass err and info to a logging service
		this.setState({ hasError: true, msg: err.message });
	}

	render() {
		if (this.state.hasError) {
			return <h1>Some error occurred.</h1>
		} else {
			// If no error occurred, then render the inner component.
			return this.props.children;
		}
	}
// This only works in Production build/mode.
```

We should not use it everywhere. Only at places where it is extremely difficult
to handle errors and it is easier to let errors occur and display respective content
to the user. Common use case could be when making the HTTP request and it fails.

More on Error-Boundaries:
https://reactjs.org/docs/error-boundaries.html
